projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -476,11 +477,22 @@ export class IgxOverlayService implements OnDestroy {
     }
 
     private setUpCloseOnEscape(info: OverlayInfo) {
+        if (!info.settings.modal) {
```

Comment:
this will not work!
If I have two overlays like this:
`Overlay1` no-modal closeOnEsc=true
`Overlay2` no-modal closeOnEsc=false
If I open `Overlay1` and then `Overlay2` I will end up with `kyedown` event attached to the `document`. If I click `ESC` key it will close `Overlay2`
Two things to fix here:
1. The check for `closeOnEsc` should be in the event handler. Each overlay should add handler with or without `closeOnEsc` set to true.
2. Check the logic in `addOutsideClickListener` and do the same.